Branch-SDK/src/io/branch/referral/network/BranchRemoteInterfaceUrlConnection.java

```diff
@@ -149,16 +150,23 @@ private BranchResponse doRestfulPost(String url, JSONObject payload, int retryNu
                 retryNumber++;
                 return doRestfulPost(url, payload, retryNumber);
             } else {
+                InputStream inputStream = null;
                 try {
+
                     if (responseCode != HttpsURLConnection.HTTP_OK && connection.getErrorStream() != null) {
-                        return new BranchResponse(getResponseString(connection.getErrorStream()), responseCode);
+                        inputStream = connection.getErrorStream();
```

Comment:
@migdol is there a particular reason you are creating a new `inputstream` for this? We are leaning towards just returning the BranchResponse and just closing the `outputStreamWriter`.